File: ConsoleApp1/model/Entity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace model
{
    [Serializable]
    public class Entity<Id>
    {
        public Id id { get; set; }

        public override string ToString()
        {
            return "Entity: {id = " + id + "}";
        }
    }
}

File: src/main/java/org/example/model/Entity.java
package org.example.model;

import java.io.Serializable;

public abstract class Entity<ID> implements Serializable {

    private ID id;
    public ID getId() {
        return id;
    }
    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public abstract String toString();
}